plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/standard/StandardDeployJob.java

```diff
@@ -0,0 +1,49 @@
+package com.google.cloud.tools.eclipse.appengine.deploy.standard;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.WorkspaceJob;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+
+import com.google.cloud.tools.eclipse.appengine.deploy.Messages;
+
+public class StandardDeployJob extends WorkspaceJob {
+  
+  private ProjectToStagingExporter projectToStagingExporter;
+  private StagingDirectoryProvider stagingDirectoryProvider;
+  private IProject project;
+
+  public StandardDeployJob(ProjectToStagingExporter exporter,
+                           StagingDirectoryProvider stagingDirectoryProvider,
+                           IProject project) {
+    super(Messages.getString("deploy.standard.runnable.name")); //$NON-NLS-1$
+    setRule(project);
+    projectToStagingExporter = exporter;
+    this.stagingDirectoryProvider = stagingDirectoryProvider;
+    this.project = project;
+  }
+
+  @Override
+  public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
+    String stageDir = stagingDirectoryProvider.get();
```

Comment:
It's simpler to just call Files.createTempDirectory here. At the least, at a TODO for this.